qabel_provider/models.py

```diff
@@ -64,6 +87,117 @@ def send_confirmation_mail(self):
         EmailAddress.objects.get_primary(self.user).send_confirmation(signup=False)
 
 
+class PlanInterval(models.Model, ExportModelOperationsMixin('planinterval')):
+    # A user may own any number of (presumably prepaid) plan intervals
+    profile = models.ForeignKey(Profile)
+    plan = models.ForeignKey(Plan)
+
+    # Duration the interval can be used
+    duration = models.DurationField()
+
+    STATES = (
+        ('pristine', 'pristine (neither expired nor used)'),
+        ('in_use', 'currently used interval'),
+        # expired := now() past (started_at + duration)
+        ('expired', 'expired interval'),
+    )
+    state = models.CharField(choices=STATES, default='pristine', max_length=10)
+
+    # Use started at <timestamp>
+    started_at = models.DateTimeField(null=True, blank=True)
+
+    def check_expiry(self):
+        """
+        Return false-ish if plan is expired, self otherwise.
+
+        Update state if expired.
+        """
+        if timezone.now() > (self.started_at + self.duration):
+            self.state = 'expired'
+            self.save()
+            audit_log = ProfilePlanLog(profile=self.profile,
+                                       action='expired-interval', plan=self.plan, interval=self)
+            audit_log.save()
+            return
+        return self
+
+    def start(self):
+        """Start using this (pristine) interval."""
+        assert self.state == 'pristine'
```

Comment:
We shouldn't use asserts that are essential for the code, use a guard close with an exception instead.

(Also, we never know if someone actually runs this with `-O`)